File: apps/java-sdk/src/main/java/com/firecrawl/models/DocumentPage.java
package com.firecrawl.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * Markdown extracted from a single PDF page.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class DocumentPage {

    private int pageNumber;
    private String markdown;

    public int getPageNumber() { return pageNumber; }
    public String getMarkdown() { return markdown; }
}

File: apps/ruby-sdk/lib/firecrawl/models/document_page.rb
# frozen_string_literal: true

module Firecrawl
  module Models
    # Markdown extracted from a single PDF page.
    class DocumentPage
      attr_reader :page_number, :markdown

      def initialize(data)
        @page_number = data["pageNumber"]
        @markdown = data["markdown"]
      end
    end
  end
end
